.github/workflows/checkstyle.yml

```diff
@@ -0,0 +1,19 @@
+name: Checkstyle
+
+on: [push, pull_request]
+
+jobs:
+  build:
+    name: Run Checkstyle
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up JDK 11
+      uses: actions/setup-java@v2
+      with:
+        java-version: '11'
+        distribution: 'adopt'
+    - name: Run Checkstyle
+      run: |
```

Comment:
This project is already set up to run checkstyle via maven. Your workflow should be similar to the existing workflows that run it, but you might need to add to the maven config to use your new rule set.